### ISSUE TITLE
pageLoadTimeout does not take effect

### ISSUE THREAD

**Issue by barancev** [mozilla/geckodriver#694]
## Firefox Version
Nightly 55.0a1 (2017-04-24) (64-bit), geckodriver 0.16.1

## Platform
Windows

## Steps to reproduce
```
WebDriver driver = new FirefoxDriver();
driver.manage().timeouts().pageLoadTimeout(2, SECONDS);
driver.get(pageThatLoads12sec);
```

TimeoutException is expected.

Trace (note the timestamps in get command trace):
```
1493281722509	geckodriver	INFO	Listening on 127.0.0.1:15595
1493281723133	webdriver::server	DEBUG	→ POST /session {
  "desiredCapabilities": {
    "nativeEvents": false,
    "firefox_binary": {
      "extraEnv": {},
      "hCode": 1881129850,
      "class": "org.openqa.selenium.firefox.FirefoxBinary",
      "timeout": 45000
    },
    "marionette": true,
    "acceptInsecureCerts": true,
    "browserName": "firefox",
    "moz:firefoxOptions": {
      "binary": "c:\\Program Files\\Nightly\\firefox.exe",
      "prefs": {},
      "args": []
    },
    "version": "",
    "platform": "ANY"
  },
  "requiredCapabilities": {},
  "capabilities": {
    "desiredCapabilities": {
      "nativeEvents": false,
      "firefox_binary": {
        "extraEnv": {},
        "hCode": 1881129850,
        "class": "org.openqa.selenium.firefox.FirefoxBinary",
        "timeout": 45000
      },
      "marionette": true,
      "acceptInsecureCerts": true,
      "browserName": "firefox",
      "moz:firefoxOptions": {
        "binary": "c:\\Program Files\\Nightly\\firefox.exe",
        "prefs": {},
        "args": []
      },
      "version": "",
      "platform": "ANY"
    },
    "requiredCapabilities": {},
    "alwaysMatch": {
      "acceptInsecureCerts": true
    },
    "firstMatch": [
      {
        "browserName": "firefox",
        "moz:firefoxOptions": {
          "binary": "c:\\Program Files\\Nightly\\firefox.exe",
          "prefs": {},
          "args": []
        }
      }
    ]
  }
}
1493281723148	geckodriver::capabilities	DEBUG	Trying to read firefox version from ini files
1493281723149	geckodriver::capabilities	DEBUG	Found version 55.0a1
1493281723163	geckodriver::marionette	INFO	Starting browser \\?\C:\Program Files\Nightly\firefox.exe with args ["-marionette"]
1493281723945	addons.manager	ERROR	startup failed: [Exception... "Component returned failure code: 0x80070057 (NS_ERROR_ILLEGAL_VALUE) [nsIFile.create]"  nsresult: "0x80070057 (NS_ERROR_ILLEGAL_VALUE)"  location: "JS frame :: resource://gre/modules/FileUtils.jsm :: FileUtils_getDir :: line 70"  data: no] Stack trace: FileUtils_getDir()@resource://gre/modules/FileUtils.jsm:70 < FileUtils_getFile()@resource://gre/modules/FileUtils.jsm:42 < validateBlocklist()@resource://gre/modules/AddonManager.jsm:653 < startup()@resource://gre/modules/AddonManager.jsm:813 < startup()@resource://gre/modules/AddonManager.jsm:3081 < observe()@jar:file://///%3F/C:/Program%20Files/Nightly/omni.ja!/components/addonManager.js:65
JavaScript error: resource://gre/modules/AddonManager.jsm, line 1593: NS_ERROR_NOT_INITIALIZED: AddonManager is not initialized
1493281724717	geckodriver::marionette	TRACE	  connection attempt 0/600
1493281725811	geckodriver::marionette	TRACE	  connection attempt 1/600
Unable to read VR Path Registry from C:\Users\alexei\AppData\Local\openvr\openvrpaths.vrpath
JavaScript error: resource://gre/modules/AddonManager.jsm, line 2508: NS_ERROR_NOT_INITIALIZED: AddonManager is not initialized
1493281726041	Marionette	DEBUG	Setting recommended pref toolkit.cosmeticAnimations.enabled to false
1493281726043	Marionette	DEBUG	Setting recommended pref datareporting.policy.dataSubmissionPolicyAccepted to false
1493281726048	Marionette	INFO	Listening on port 62993
1493281726415	geckodriver::marionette	DEBUG	Connected to Marionette on localhost:62993
1493281726419	Marionette	DEBUG	Accepted connection 0 from 127.0.0.1:63009
1493281726429	geckodriver::marionette	TRACE	← {"applicationType":"gecko","marionetteProtocol":3}
1493281726429	geckodriver::marionette	TRACE	→ 163:[0,1,"newSession",{"acceptInsecureCerts":true,"browserName":"firefox","capabilities":{"desiredCapabilities":{"acceptInsecureCerts":true,"browserName":"firefox"}}}]
1493281726437	Marionette	TRACE	0 -> [0,1,"newSession",{"acceptInsecureCerts":true,"browserName":"firefox","capabilities":{"desiredCapabilities":{"acceptInsecureCerts":true,"browserName":"firefox"}}}]
1493281726440	Marionette	CONFIG	Matched capabilities: {"browserName":"firefox","browserVersion":"55.0a1","platformName":"windows_nt","platformVersion":"6.1","pageLoadStrategy":"normal","acceptInsecureCerts":true,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"rotatable":false,"specificationLevel":0,"moz:processID":14832,"moz:profile":"C:\\Users\\alexei\\AppData\\Local\\Temp\\rust_mozprofile.qTT6feLVpcpo","moz:accessibilityChecks":false}
1493281726440	Marionette	WARN	TLS certificate errors will be ignored for this session
1493281726524	Marionette	TRACE	0 <- [1,1,null,{"sessionId":"e4ce9019-3261-45d3-8537-648ca43d1957","capabilities":{"browserName":"firefox","browserVersion":"55.0a1","platformName":"windows_nt","platformVersion":"6.1","pageLoadStrategy":"normal","acceptInsecureCerts":true,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"rotatable":false,"specificationLevel":0,"moz:processID":14832,"moz:profile":"C:\\Users\\alexei\\AppData\\Local\\Temp\\rust_mozprofile.qTT6feLVpcpo","moz:accessibilityChecks":false}}]
1493281726528	geckodriver::marionette	TRACE	← [1,1,null,{"sessionId":"e4ce9019-3261-45d3-8537-648ca43d1957","capabilities":{"browserName":"firefox","browserVersion":"55.0a1","platformName":"windows_nt","platformVersion":"6.1","pageLoadStrategy":"normal","acceptInsecureCerts":true,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"rotatable":false,"specificationLevel":0,"moz:processID":14832,"moz:profile":"C:\\Users\\alexei\\AppData\\Local\\Temp\\rust_mozprofile.qTT6feLVpcpo","moz:accessibilityChecks":false}}]
1493281726528	webdriver::server	DEBUG	← 200 OK {"value": {"sessionId":"e4ce9019-3261-45d3-8537-648ca43d1957","capabilities":{"acceptInsecureCerts":true,"browserName":"firefox","browserVersion":"55.0a1","moz:accessibilityChecks":false,"moz:processID":14832,"moz:profile":"C:\\Users\\alexei\\AppData\\Local\\Temp\\rust_mozprofile.qTT6feLVpcpo","pageLoadStrategy":"normal","platformName":"windows_nt","platformVersion":"6.1","rotatable":false,"specificationLevel":0,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000}}}}

1493281727304	webdriver::server	DEBUG	→ POST /session/e4ce9019-3261-45d3-8537-648ca43d1957/timeouts {"page load":2000}
1493281727304	geckodriver::marionette	TRACE	→ 19:[0,2,"timeouts",{}]
1493281727306	Marionette	TRACE	0 -> [0,2,"timeouts",{}]
1493281727307	Marionette	TRACE	0 <- [1,2,null,{}]
1493281727308	geckodriver::marionette	TRACE	← [1,2,null,{}]
1493281727308	webdriver::server	DEBUG	← 200 OK {"value": {}}

1493281727317	webdriver::server	DEBUG	→ POST /session/e4ce9019-3261-45d3-8537-648ca43d1957/url {"url":"http://comp1:31637/common/sleep?time=12"}
1493281727317	geckodriver::marionette	TRACE	→ 61:[0,3,"get",{"url":"http://comp1:31637/common/sleep?time=12"}]
1493281727318	Marionette	TRACE	0 -> [0,3,"get",{"url":"http://comp1:31637/common/sleep?time=12"}]
1493281739698	Marionette	TRACE	0 <- [1,3,null,{}]
1493281739703	geckodriver::marionette	TRACE	← [1,3,null,{}]
1493281739703	webdriver::server	DEBUG	← 200 OK {"value": {}}
```

**Comment by andreastt** [comment]
You’re setting the page load timeout to 5 minutes (300000 ms) and the document loads after 12 (I assume) seconds, because the call you make to set it to 2000 ms doesn’t use the correct input:

    1493281727304	webdriver::server	DEBUG	→ POST /session/e4ce9019-3261-45d3-8537-648ca43d1957/timeouts {"page load":2000}

This key is supposed to be `pageLoad`.

**Comment by andreastt** [comment]
Can you link to the issue/commit when you file/fix it? We’ve had a lot of reports about this not working, and I’ve discarded them, so it would be good to link them to that.

**Comment by barancev** [comment]
Fixed by https://github.com/SeleniumHQ/selenium/commit/91f9f2a6194a9c33006e9415c1f257c6a1fa5c04 , should be released in Selenium 3.5
